File: mobile/lib/features/trusted_contacts/domain/trusted_contact.dart
class TrustedContact {
  final int? id;
  final String name;
  final String phone;
  final String? relation;
  final bool allowSessionAlerts;
  final bool allowLostPhoneAlerts;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  TrustedContact({
    this.id,
    required this.name,
    required this.phone,
    this.relation,
    this.allowSessionAlerts = false,
    this.allowLostPhoneAlerts = false,
    this.createdAt,
    this.updatedAt,
  });

  factory TrustedContact.fromJson(Map<String, dynamic> json) {
    return TrustedContact(
      id: json['id'],
      name: json['name'],
      phone: json['phone'],
      relation: json['relation'],
      allowSessionAlerts: json['allow_session_alerts'] ?? false,
      allowLostPhoneAlerts: json['allow_lost_phone_alerts'] ?? false,
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : null,
      updatedAt: json['updated_at'] != null ? DateTime.parse(json['updated_at']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'name': name,
      'phone': phone,
      'relation': relation,
      'allow_session_alerts': allowSessionAlerts,
      'allow_lost_phone_alerts': allowLostPhoneAlerts,
    };
  }

  TrustedContact copyWith({
    int? id,
    String? name,
    String? phone,
    String? relation,
    bool? allowSessionAlerts,
    bool? allowLostPhoneAlerts,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return TrustedContact(
      id: id ?? this.id,
      name: name ?? this.name,
      phone: phone ?? this.phone,
      relation: relation ?? this.relation,
      allowSessionAlerts: allowSessionAlerts ?? this.allowSessionAlerts,
      allowLostPhoneAlerts: allowLostPhoneAlerts ?? this.allowLostPhoneAlerts,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: backend/app/models/trusted_contact.py
from datetime import datetime
from datetime import timezone
from typing import Optional

from sqlalchemy import ForeignKey, String, Boolean, DateTime, Integer
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.session import Base

class TrustedContact(Base):
    __tablename__ = "trusted_contacts"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), index=True)
    name: Mapped[str] = mapped_column(String)
    phone: Mapped[str] = mapped_column(String)
    relation: Mapped[Optional[str]] = mapped_column(String)
    allow_session_alerts: Mapped[bool] = mapped_column(Boolean, default=False)
    allow_lost_phone_alerts: Mapped[bool] = mapped_column(Boolean, default=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))

    owner = relationship("User", back_populates="trusted_contacts")
